lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { FocusEvent, KeyboardEvent, Ref, useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+import { always, dec, inc } from 'ramda';
+
+import './NxCombobox.scss';
+
+import forwardRef from '../../util/genericForwardRef';
+import { Props, propTypes } from './types';
+import NxTextInput from '../NxTextInput/NxTextInput';
+import NxDropdownMenu from '../NxDropdownMenu/NxDropdownMenu';
+import NxLoadError from '../NxLoadError/NxLoadError';
+import NxLoadingSpinner from '../NxLoadingSpinner/NxLoadingSpinner';
+import { useUniqueId } from '../../util/idUtil';
+import DataItem from '../../util/DataItem';
+export { Props } from './types';
+
+function NxComboboxRender<T extends string | number = string>(
+  props: Props<T>,
+  ref: Ref<HTMLDivElement>
+) {
+  const {
+        className: classNameProp,
+        loading,
+        loadError,
+        matches,
+        value,
+        onChange,
+        onSearch,
+        disabled,
+        emptyMessage,
+        autoComplete,
+        validatable,
+        isPristine,
+        validationErrors,
+        autoSelectedVal,
```

Comment:
I'd think we'd want `autoSelectedVal` to be managed totally internally to the component - have the `useState` for it be in here rather than in the calling code. Actually it might not even need to be its own state, it can probably be derived on the fly from `value`, `matches`, and `inlineStyle`.  Or alternatively, maybe use state for `autoSelectedVal` and get rid of `inlineStyle`.

Reply:
Thanks for the suggestion, I have updated it using `inlineStyle` here 99db2e2175a28b22fff0bddbf0fe57cb896a7761, thanks.